atomic_reactor/tasks/binary_container_build.py

```diff
@@ -263,20 +266,21 @@ def build_container(
         # passing stdout=PIPE guarantees that stdout is not None, but the type hints for the
         #   subprocess module do not express that (TL;DR - this is just for type checkers)
         assert build_process.stdout is not None
+        assert build_process.stderr is not None
 
-        last_line = None
+        error_lines = []
 
-        while True:
-            if line := build_process.stdout.readline():
-                yield line
-                # keep the last non-empty, non-whitespace line for an eventual error message
-                last_line = line.rstrip() or last_line
+        for line in build_process.stdout:
+            yield False, line
+        for line in build_process.stderr:
+            if line.rstrip():
+                error_lines.append(line)
+            yield True, line
 
-            if (rc := build_process.poll()) is not None:
-                break
+        rc = build_process.poll()
 
         if rc != 0:
-            error = last_line if last_line else "<no output!>"
+            error = '\n'.join(error_lines) if error_lines else "<no output!>"
             raise BuildProcessError(f"Build failed (rc={rc}): {error}")
```

Comment:
IMO, the aim of story was to just log this exception also as ERROR log line, to have it in log aggregation as ERROR

Reply:
updated to use last line as error only.